NExtends/Primitives/String.extensions.cs

```diff
@@ -9,555 +9,577 @@
+            {
+                //Ie, of course
+                //http://stackoverflow.com/questions/1877788/javascript-date-to-c-sharp-via-ajax pour ie
+                result = DateTime.ParseExact(date, "ddd MMM d HH:mm:ss UTCzzzzz yyyy", CultureInfo.InvariantCulture);
+            }
+            return result;
+        }
+
+        public static string Format(this string s, params object[] args)
+        {
+            return args != null ? string.Format(s, args) : s;
+        }
+
+        //Validation des GUID qui arrivent en paramètre
+        static Regex isGuid = new Regex(@"^(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}$", RegexOptions.Compiled);
+
+        //Validation des numéro de sécu INSEE
+        //version de gabsoftware
+        //http://www.developpez.net/forums/d677820/php/langage/regex/verification-numero-securite-sociale/
+        private const string REGULAR_INSEE_NUMBER = @"([1-3])[\s\.\-]?(\d{2})[\s\.\-]?(0\d|[2-35-9]\d|[14][0-2])[\s\.\-]?(0[1-9]|[1-9]\d|2[abAB])[\s\.\-]?(\d{3})[\s\.\-]?(\d{3})[\s\.\-\|]{0,3}";
+        private const string TEMPORARY_INSEE_NUMBER = @"(?<temp>[7-9][\s\.\-]?\d{2}[\s\.\-]?\d{2}[\s\.\-]?\d{2}[\s\.\-]?\d{3}[\s\.\-]?\d{3}[\s\.\-\|]?)";
+        private const string KEY_INSEE_NUMBER = @"(?<key>[0-8]\d|9[0-7])?";
+
+        static Regex isINSEENumber = new Regex(String.Format(@"^(({0})|({1})){2}$", REGULAR_INSEE_NUMBER, TEMPORARY_INSEE_NUMBER, KEY_INSEE_NUMBER), RegexOptions.Compiled);
+
+
+        public static bool IsGuid(string candidate)
+        {
+            return (!string.IsNullOrEmpty(candidate) && isGuid.IsMatch(candidate));
+        }
+
+        public static bool IsINSEENumber(this string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return false;
+            }
+
+            var match = isINSEENumber.Match(input);
+            if (!match.Success)
+            {
+                return false;
+            }
+
+            if (!string.IsNullOrEmpty(match.Groups["key"].Value))
+            {
+                var number = Int64.Parse(string.Join("", input.Split(' ', '.', '-', '|')).Replace("2a", "19", StringComparison.OrdinalIgnoreCase).Replace("2b", "18", StringComparison.OrdinalIgnoreCase));
+                var rest = ((number / 100) + number % 100) % 97;
+                if (rest != 0)
+                {
+                    return false;
+                }
+            }
+            else if (!string.IsNullOrEmpty(match.Groups["temp"].Value))
+            {
+                // Si le numéro est de type temporaire, on impose la clé (donc ça doit passer dans le if au-dessus)
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// On passe par un MailAddress, plus simple
+        /// </summary>
+        public static bool isEmail(this string email)
```

Comment:
It could be nice to introduce IsEmail with i capital letter, and obsolete this one. (code convention)